File: 232_Implement_Queue_using_stack/Queue.py
class Queue(object):
    def __init__(self):
        """
        initialize your data structure here.
        """
        self.items = []

    def push(self, x):
        """
        :type x: int
        :rtype: nothing
        """
        self.items.insert(0, x)

    def pop(self):
        """
        :rtype: void
        """
        self.items.pop()


    def peek(self):
        """
        :rtype: int
        """
        return self.items[-1]

    def empty(self):
        """
        :rtype: bool
        """
        return not self.items

q = Queue()
q.push(1)
q.push(2)
q.push(3)
q.pop()
print q.peek()

File: 232_Implement_Queue_using_stack/Queue.cpp
#include <stack>

using namespace std;

class Queue {
    public:
        void push(int x) {
            _input.push(x);
        }

        void pop() {
            peek();
            _output.pop();
        }

        int peek() {
            if(_output.empty()) {
                while(_input.size()) {
                    _output.push(_input.top());
                    _input.pop();
                }
            }
            return _output.top();
        }

        bool empty() {
            return _input.empty() && _output.empty();
        }
    private:
        stack<int> _input;
        stack<int> _output;
};

int main() {
    return 0;
}
